File: benchmarks/bench.ts
// Typescript script to run and compare the benchmarks for eslint and rslint.

import {
  bench,
  BenchmarkTimer,
  runBenchmarks,
} from "https://deno.land/std@0.88.0/testing/bench.ts";

// number of times that each project should linted to prevent variance.
const RUN_COUNT = 20;

// A list of projects that should be linted.
const PROJECTS = ["benchmarks/projects/oak", "benchmarks/projects/engine262"];

PROJECTS.forEach((project) => {
  bench({
    name: `rslint ${project}`,
    runs: RUN_COUNT,
    async func(b: BenchmarkTimer): Promise<void> {
      b.start();
      const proc = Deno.run({
        cmd: ["./target/release/rslint", project],
        stdout: "null",
        stderr: "null",
      });

      await proc.status();
      b.stop();
    },
  });

  bench({
    name: `eslint ${project}`,
    runs: RUN_COUNT,
    async func(b: BenchmarkTimer): Promise<void> {
      b.start();
      const proc = Deno.run({
        cmd: [
          "npm",
          "run",
          "eslint",
          "--",
          "--no-eslintrc",
          "--ext",
          ".js",
          "--ext",
          ".ts",
          project,
        ],
        cwd: "./benchmarks",
        stdout: "null",
        stderr: "null",
      });

      await proc.status();
      b.stop();
    },
  });
});

const data = await runBenchmarks();
console.log(data);

File: benchmarks/bench.py
import subprocess

def eslint_bench(project):
    project = "projects/" + project
    subprocess.run([
        "npm",
        "run",
        "eslint",
        "--",
        "--no-eslintrc",
        "--ext",
        ".js",
        "--ext",
        ".ts",
        project,
    ], cwd="./benchmarks")

def rslint_bench(project):
    project = "projects/" + project
    subprocess.run([ "../target/release/rslint", project ], cwd="./benchmarks")

def test_eslint_oak(benchmark):
    benchmark(eslint_bench, "oak")

def test_rslint_oak(benchmark):
    benchmark(rslint_bench, "oak")

def test_eslint_engine262(benchmark):
    benchmark(eslint_bench, "engine262")

def test_rslint_engine262(benchmark):
    benchmark(rslint_bench, "engine262")
